addon/index.js

```diff
@@ -426,4 +415,21 @@ AddonManager.addInstallListener({
   onDownloadFailed: function(install) {
     app.send('addon-install:download-failed', formatInstallData(install));
   }
+};
+AddonManager.addInstallListener(installListener);
```

Comment:
Could we collect these `add*Listener` methods into some kind of addon startup method? There's some loose initialization code around lines 126-141, too. Would be great to have a single, scannable place to grok the startup sequence, and I don't think it'd be _too_ much creep for this PR.

The `unload.when` callback below looks like a great candidate to extract out into a top-level shutdown method (yay symmetry).